### ISSUE TITLE
asyncdispatch.poll() eats cpu; poll is not blocking

### ISSUE THREAD

**Issue by enthus1ast** [nim-lang/Nim#6929]
```nim
import asyncdispatch
proc foo() {.async.} = 
  while true:
    echo "in foo"
    await sleepAsync(1000)

asyncCheck foo()
while true:
  echo "."
  poll(500)
```

results in 100% cpu load on one core. Tested on linux

edit: fixed typo

**Comment by dom96** [comment]
Proves we need to move to proper timers :)

**Comment by cheatfate** [comment]
All latest changes to `poll()` procedure causes this bug... And changing behavior of `sleepAsync()` to use hardware timers is not a solution for this problem.

**Comment by cheatfate** [comment]
After some investigation, i have found that this bug is regression, made by somebody who introduced this branch (https://github.com/nim-lang/Nim/blob/devel/lib/pure/asyncdispatch.nim#L1234).
As i mentioned in other issues, this happens just because of policy of rash and untested 'improvements'.

@enthus1ast, remove this line, and everything will be fine.

**Comment by dom96** [comment]
Are you sure this is a regression? This line was always in there: https://github.com/nim-lang/Nim/blame/000b8afd26fa16684a116d9afe798ea94df9c270/lib/upcoming/asyncdispatch.nim#L1222

**Comment by cheatfate** [comment]
@dom96, i'm sure that exactly this check must be removed, but this will enable #4262 again, but #4262 must be fixed in other way.

**Comment by enthus1ast** [comment]
this is the output of the above test with 
asyncdispatch.nim#L1222 removed:

```nim
.
.
in foo
.
.
.
.
.
.
.
in foo
.
.
in foo
.
.
in foo
.
.
.
.
.
.
.
.
.
.
.
in foo
.
.
in foo
.
```

**Comment by cheatfate** [comment]
@enthus1ast, looks like work as expected, isn't it?

**Comment by enthus1ast** [comment]
it works! (i just have no clue how asyncdispatch works inside, i guess...)

**Comment by dom96** [comment]
You were right all along @cheatfate :)

I decided to finally investigate how to fix #7758/#7886 and ended up doing exactly what you suggest here. I can't reproduce #4262 anymore though.
